### PR TITLE
Initial snap in

### PR DESCRIPTION
**Pull request by saary**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+const { JsonFilter } = require('jsfilter');
+
+const mergeBits = ({ subject, bits }) => (
+  bits
+    .filter((bit) => {
+      // only bits which criteria matches the subject should remain
+      if (bit.criteria) {
```
Comment (Vall3y) [inline index.js]: I prefer reversing the if in these spots (`!if (!bit.criteria) return;`)

File: index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    .reduce((res, bit) => Object.assign(res, bit.data), {})
+);
+
+module.exports = ({ bits = [] }) => {
+  return {
+    bits,
+    set(bits) {
+      this.bits = bits;
+    },
+    get(subject) {
```
Comment (Vall3y) [inline index.js:32]: subject is kind of a weird name? This seems more like a query

**Review by Vall3y — APPROVED** [review]

**Comment by Vall3y** [comment]
Very Cool

File: index.js (hunk 2 above)
Comment (saary) [reply index.js:32]: Was struggling to find a good name.
It is the object we apply the criteria to.

Does query fit that definition?

Comment (Vall3y) [reply index.js:32]: You're querying the confbit object to give you a config that answers your query, is the way I was looking at it

**Comment by saary** [comment]
Sounds good, I will switch

File: index.js (hunk 2 above)
Comment (saary) [reply index.js:32]: Sounds good, I will switch

**Pull request merged**